### PR TITLE
feat: #CRRE-486 filter orders selection through some admin actions

### PR DESCRIPTION
**Pull request by sjollois**
## Describe your changes
Filter sent or refused orders when the admin refused, validate or sent the orders to the library. Disable the buttons "send to the library" when the filter "sent to the library" is checked
## Checklist tests
Check "Sent to the library" filter and the buttons must be disabled.
if you try to sent orders that already be"sent to the library", they will be filtered.
## Issue ticket number and link
[https://jira.support-ent.fr/browse/CRRE-486](url)
## Checklist before requesting a review (magic string, indentation, comment/documentation...)

- [x] I have detailed the tests to do in my feature/fix in order to prevent consequents regressions (must specify in **Checklist tests**)
- [x] I have performed a self-review of my code
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation (API Doc etc...) - (must specify in **Description** for target version)
- [ ] If it is a consequent feature, I have added thorough tests.
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] Any dependent changes have been added to this project (must specify in **Description**)

### CODE REVIEW

File: src/main/resources/public/template/administrator/order/order-waiting.html [hunk 1]
```diff
@@ -199,10 +200,21 @@ <h3><i18n>crre.info.empty.tab</i18n></h3>
...
+                    </button>
+                    <button ng-click="openRefusingOrderLightbox()"
+                            ng-disabled="checkStatusChoice('SENT')">
+                        <i18n>Denied</i18n>
+                    </button>
+                    <button ng-click="exportCSVRegion(false, display.allOrdersSelected) && uncheckAll()">
+                        <i18n>crre.logs.csv.export</i18n>
+                    </button>
+                    <button ng-click="openConfirmGenerateLibraryLightbox()"
+                            ng-disabled="checkStatusChoice('SENT')">
```
Comment (Rseuret) [inline src/main/resources/public/template/administrator/order/order-waiting.html:215]: t'as une enum plus loin avec la liste des statuts tu pourrais l'utiliser ici non ?

**Review by Rseuret — CHANGES_REQUESTED** [review]

File: src/main/resources/public/template/administrator/order/order-waiting.html (hunk 1 above)
Comment (sjollois) [reply src/main/resources/public/template/administrator/order/order-waiting.html:215]: mais il ne le prend pas en compte dans le HTML ou alors tu as une idée de comment faire ?

Comment (Rseuret) [reply src/main/resources/public/template/administrator/order/order-waiting.html:215]: tu peux le mettre dans une constant dans ton controller et l'appeler dans l html je pense

File: src/main/resources/public/ts/model/OrderRegion.ts [hunk 2]
```diff
@@ -167,11 +168,13 @@ export class OrdersRegion extends Selection<OrderRegion> {
...
+            const filteredOrder : OrderRegion[] = this.all
+                .filter(order => order.status != ORDER_STATUS_ENUM.SENT);
             let params_id_equipment = new Set();
             let params_id_structure = new Set();
-            let params_id_order = this.all.map(order => order.id);
-            this.all.forEach(order => params_id_equipment.add(order.equipment_key));
-            this.all.forEach(order => params_id_structure.add(order.id_structure));
+            let params_id_order = filteredOrder.map(order => order.id);
+            filteredOrder.forEach(order => params_id_equipment.add(order.equipment_key));
+            filteredOrder.forEach(order => params_id_structure.add(order.id_structure));
```
Comment (BastienBenard) [inline src/main/resources/public/ts/model/OrderRegion.ts:177]: typage du paramètre order dans les lambda

**Review by BastienBenard — APPROVED** [review]

**Review by Sofianebernoussi — APPROVED** [review]

**Pull request merged**